### DISCUSSION TITLE
Firedrake Installation on Mac - Issues with h5py or PETSc

### DISCUSSION THREAD

**Discussion by sorenandflying** [firedrakeproject/firedrake#3021] [Firedrake support] [answered]
Hi All,

I am trying to install Firedrake and Icepack on a 2020 macbook air (Ventura 13.4, python 3.11.4).

I've resolved a few issues with PETSc by following the instructions on [this post](https://github.com/ldeo-glaciology/LEAP-Cryo-planning/blob/main/icepack_install.md) and [this post](https://github.com/firedrakeproject/firedrake/discussions/2851), but am still running into some errors during installation.

Some familiar warnings are popping up in the logs, but as far as I can tell I've updated xcode, installed updated python (I've tried updating the path and using the `(brew--prefix)`), and am using a version of firedrake-install from the last few days.

`dylib (/usr/local/opt/openblas/lib/libopenblas.dylib) was built for newer macOS version (13.0) than being linked (11.0)`

The failure point is when trying to build h5py, running into some errors when building the wheel.

Any ideas for next steps would be super appreciated.

Install log:
[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/12021281/firedrake-install.log)

PETSc logs:
[configure.log](https://github.com/firedrakeproject/firedrake/files/12021295/configure.log)
[make.log](https://github.com/firedrakeproject/firedrake/files/12021296/make.log)

**Comment by connorjward** [comment, marked as answer]
I am seeing warnings like:
```
  ld: warning: dylib (/Users/sorenthompson/firedrake/src/petsc/default/lib/libhdf5.dylib) was built for newer macOS version (11.0) than being linked (10.9)
```
This is surprising because the lowest these values should be is, I think, 11.0 ([source](https://github.com/firedrakeproject/firedrake/blob/master/scripts/firedrake-install#L1154)). Is your `firedrake-install` definitely up-to-date and is this the right log file?

This log is also saying that you are not using the Homebrew Python. Could you try running `$(brew --prefix)/bin/python3 firedrake-install <other args>` to force the right one?

**Reply by sorenandflying** [reply]
I'm not entirely sure what changed, but it looks like clearing out everything and running it again got a successful installation. Thanks!

After installation was complete, I was having some issues importing firedrake into my notebooks (getting the `no module named 'firedrake'` error). Installing a copy Jupyter Notebook within the venv seemed to fix this for me.

**Reply by dham** [reply]
Jupyter notebooks are just running *a* Python interpreter. In order to find Firedrake, that Python interpreter needs to be the one in the venv where Firedrake is installed. Installing Jupyter notebook in the venv is one way to achieve that.
